tests/composite_pk/test_filter.py

```diff
@@ -407,3 +408,51 @@ def test_cannot_cast_pk(self):
         msg = "Cast does not support composite primary keys."
         with self.assertRaisesMessage(ValueError, msg):
             Comment.objects.filter(text__gt=Cast(F("pk"), TextField())).count()
+
+    def test_outer_ref_pk(self):
+        subquery = Subquery(Comment.objects.filter(pk=OuterRef("pk")).values("id"))
+        tests = [
+            ("", 5),
+            ("__gt", 0),
+            ("__gte", 5),
+            ("__lt", 0),
+            ("__lte", 5),
+        ]
+        for lookup, expected_count in tests:
+            with self.subTest(f"id{lookup}"):
+                queryset = Comment.objects.filter(**{f"id{lookup}": subquery})
+                self.assertEqual(queryset.count(), expected_count)
+
+    def test_non_outer_ref_subquery(self):
+        # If rhs is any non-OuterRef object with an as_sql() function.
+        pk = Exact(F("tenant_id"), 1)
+        msg = (
+            "'exact' subquery lookup of 'pk' only supports OuterRef objects "
+            "(received 'Exact')"
+        )
+        with self.assertRaisesMessage(ValueError, msg):
+            Comment.objects.filter(pk=pk)
+
+    def test_outer_ref_not_composite_pk(self):
+        subquery = Comment.objects.filter(pk=OuterRef("id")).values("id")
+        queryset = Comment.objects.filter(id=Subquery(subquery))
+
+        msg = "Composite field lookups only work with composite expressions."
+        with self.assertRaisesMessage(ValueError, msg):
+            self.assertEqual(queryset.count(), 5)
+
+    def test_outer_ref_in_filtered_relation(self):
+        msg = (
+            "This queryset contains a reference to an outer query and may only be used "
+            "in a subquery."
+        )
+        with self.assertRaisesMessage(ValueError, msg):
```

Comment:
This test causes a recursion error on Oracle; it's at the origin of the broken CI

https://djangoci.com/job/django-oracle/1468/

Reply:
No worries @csirmazbendeguz! I agree that it would be great if were able to systematically run Oracle tests but this is unfortunately not feasible because it's so much slower than other backends to run the suite at ~1h40 minutes.

We've had discussions with Oracle in the past for them to provide some instance that could be used for CI purpose but it never materialized. I have hope that the recent changes in https://github.com/django/django-docker-box will bring us closer to a point where we can run things on Github actions but in the mean time the rule of thumb I use to invoke Oracle tests is whether or not any aspect touches SQL compilation. When it's the case I systematically invoke the Oracle suite.